### ISSUE TITLE
jax.scipy.stats.logistic.logpdf overflows easily with negative input

### ISSUE THREAD

**Issue by kluopaja** [google/jax#10219]
Running
```python
import jax
print(jax.scipy.stats.logistic.logpdf(-100.0))
print(jax.scipy.stats.logistic.logpdf(100.0))
```
outputs
```python
-inf
-100.0
```
which is unexpected since the logistic distribution should be symmetric.

The reason for this is the overflowing `exp` in the implementation:
https://github.com/google/jax/blob/35b32eef96e579e5f565c48f5962463f8d82192d/jax/_src/scipy/stats/logistic.py#L22-L24

An easy fix would be taking the absolute value of `x` at the start of the function but I don't know if this could introduce some problems with `jax.grad`.

**Comment by YouJiacheng** [comment]
What about
```python
import jax
def logpdf(x): 
    return - 2. * jax.numpy.logaddexp(x / 2, lax.neg(x) / 2)
```

**Comment by froystig** [comment]
As I understand it, this was fixed by #10220.
